Signup: A/B test the headline for Domains step

klimeryk opened the pull request:
The hypothesis is that removing the word "Domain" from the headline on step 3 will increase signup rate by 1%.

P2 post TBD.
@Automattic/editorial - the updated version might be a bit rough, could you please have a look at it? The idea is to avoid using the word `domain` so that the user is not put off by it.

`original` variation:
<img width="1029" alt="screen shot 2017-03-07 at 20 42 51" src="https://cloud.githubusercontent.com/assets/3392497/23674772/57d72214-0377-11e7-9a1d-2387c313d9fc.png">

`updated` variation:
<img width="935" alt="screen shot 2017-03-09 at 22 55 58" src="https://cloud.githubusercontent.com/assets/3392497/23772544/9b4112ee-051b-11e7-9410-fa71ac6c3146.png">


### Testing
Try to create a new site, hit the Domains step and verify the correct copy is used.

klimeryk commented on client/lib/abtest/active-tests.js:
```diff
@@ -104,5 +104,13 @@ module.exports = {
+			updated: 50
+		},
+		defaultVariation: 'original',
+		allowExistingUsers: true
```
@lucasartoni Should we allow existing users to participate in the test (as it is now) or not?

klimeryk commented on client/lib/abtest/active-tests.js:
```diff
@@ -104,5 +104,13 @@ module.exports = {
 	},
-
+	signupDomainsHeadline: {
+		datestamp: '20170306',
```
Will update to `20170313` (the planned date we start this test) before merging.

klimeryk commented on client/lib/abtest/active-tests.js:
```diff
@@ -104,5 +104,13 @@ module.exports = {
 	},
-
+	signupDomainsHeadline: {
+		datestamp: '20000313',
```
Will update to `20170313` (planned date the test starts) before merging.

klimeryk commented on client/signup/steps/domains/index.jsx:
```diff
@@ -251,8 +262,8 @@ const DomainsStep = React.createClass( {
 				positionInFlow={ this.props.positionInFlow }
 				signupProgress={ this.props.signupProgress }
 				subHeaderText={ this.translate( 'First up, let\'s find a domain.' ) }
```
AFAIK, this copy is only used when this step is shown as the first step in signup - so `/start/domain-first`. But that's not enabled for any production env, so we shouldn't worry about it.

michelleweber commented:
My inclination is to keep it simpler and direct, to stay away from things like cool/compelling (which can create a lot of pressure for people), and to use the language that they'll see in these settings later -- "site title." Eg:

> **Pick a title for your new site**
> We'll use the title you choose to automatically create your new site's address

It seems like you'll also want to change the text that appears in the search box to something like "Enter a title, or a few words to describe your site."

Does the note underneath re: domains need to change as well?

aidvu commented on client/signup/steps/domains/index.jsx:
```diff
@@ -243,6 +244,16 @@ const DomainsStep = React.createClass( {
+		if ( abtest( 'signupDomainsHeadline' ) === 'updated' ) {
+			fallbackHeaderText = this.translate( 'Let\'s find a cool name for your brand new site' ),
+			fallbackSubHeaderText = this.translate( 'Choose a compelling name, it will be the address of your new site!' );
+		} else {
```
You can probably skip the else and just assign defaults above, and change on `A/B` test. Your call.

klimeryk replied:
I prefer this pattern as it makes more sense to me - I assign either value based on a condition.
While
```
let foo = 'bar';
if ( something ) {
  foo = 'lol, nope';
}
```
seems confusing to me - if I can immediately check if a different value should be assigned to this variable, why assign the original value at all?

klimeryk commented:
@michelleweber Thanks for looking at this!

> **Pick a title for your new site**
> We'll use the title you choose to automatically create your new site's address

I'm not sure `title` is the best fit here - from a technical point of view, the user needs to choose either their new `.wordpress.com` subdomain or a domain to register/map. So based on the word `title`, I wouldn't try to input, for example, `example.com`. While ideally we should show relevant and interesting suggestions based on their site's title, I'm not sure we're there yet.
On the other hand, I like how it's been simplified and put in terms the user might be familiar with (and/or will see later on).

> It seems like you'll also want to change the text that appears in the search box to something like "Enter a title, or a few words to describe your site."

I'm not sure - @lucasartoni, we talked about the headings, but what about the placeholder for the domain search field?
<img width="900" alt="screen shot 2017-03-07 at 23 01 32" src="https://cloud.githubusercontent.com/assets/3392497/23680026/07e0ff9c-038a-11e7-9362-b2b2356bc45c.png">

> Does the note underneath re: domains need to change as well?

Hmmm, if we'd get rid of all the other mentions of `domain` on this step, then this visual aid could probably go too. Though, I find it a great way of explaining what a domain is 🤔 
A quick mockup of how that'd look (with the placeholder empty - not sure what to put there now):
<img width="946" alt="screen shot 2017-03-07 at 23 06 17" src="https://cloud.githubusercontent.com/assets/3392497/23680227/be5591c0-038a-11e7-8ab9-5d7caa2985dd.png">

michelleweber commented:
Okay, I think I get it -- I was going off the "cool name" in the mockup and the option to just enter a keyword, which I thought was suggesting that you want me to just put in what I want my site to be called, ie, the title. Make sense? If that's _not_ the case, then I'd suggest "address." Is this more along the line you were thinking of?

> **Let's give your site an address**
> Enter a name or some keywords, and we'll use them to create your new site's address. 
> 

(Aside: Do we know what people are typing into this box now? Are people actually typing in somethingsomething.com, or just words/a title?)

I agree that the little "already own a domain" bit is helpful, but would suggest that you keep only the link and omit the text note. If people are savvy enough to already own a domain, they'll click; if not, they don't need more text/terminology to distract them right now.

klimeryk commented:
> Okay, I think I get it -- I was going off the "cool name" in the mockup and the option to just enter a keyword, which I thought was suggesting that you want me to just put in what I want my site to be called, ie, the title.

Ideally, that's how it should work :) We (Team Cobalt on Spectrum) will be improving domain search as our main focus this year, so hopefully we'll reach that point soon. But right now, the suggestions usually take all the keywords and try to use them in a domain. So if you input here "Random Thoughts About Paleontology", it will suggest domains like `randomthoughtsaboutpaleontology.blog`... while there are possbily better options available, like `randompaleontology.blog`, etc. So that's why we'd prefer the user to input keywords or an exact domain name _for now_.
So this copy:

> **Let's give your site an address**
> Enter a name or some keywords, and we'll use them to create your new site's address.

fits great and I like it very much 👍 I've updated the PR and the screenshot in the original post with this copy. I've also updated slightly the placeholder to say `Enter **name** or keyword`, instead of `Enter domain or keyword`, since it seemed to flow better with the copy in the header. Then, the bit below explains what a `domain name` is, in case the user was still confused. It's not perfect, but this test is just a warm-up - we want to set up a streamlined process first for A/B testing, before tackling more complex A/B tests. Hence why I want to keep the changes in this test as small as possible - in future tests we can investigate completely removing mentioning/explaining the domains here 👍

> (Aside: Do we know what people are typing into this box now? Are people actually typing in somethingsomething.com, or just words/a title?)

For what I recall, most clickthroughs come when users input directly the domain they want. So, obviously, our suggestions are not that spot on (though, they might lead the user to input a domain they like). As mentioned above, we'll be working on improving them :man_dancing:

michelleweber commented:
One small thing I'd like to change -- in the explanatory text now seems insufficiently clear as I look at the mockup. Whose name? My name? What keywords? So I'm suggesting:

> Enter your site's name, or some key words that describe it -- we'll use this to create your new site's address.

Clarity above all :)

Otherwise, looks good!

klimeryk commented:
Perfection! PR and screenshots updated. Thanks again for your input and help, @michelleweber! 🙇

klimeryk commented on client/components/domains/register-domain-step/index.jsx:
```diff
@@ -220,7 +221,10 @@ const RegisterDomainStep = React.createClass( {
 	render: function() {
-		const queryObject = getQueryObject( this.props );
+		const queryObject = getQueryObject( this.props ),
+			placeholder = ( this.props.isSignupStep && abtest( 'signupDomainsHeadline' ) === 'updated' )
```
I've added the `this.props.isSignupStep` so that users that use domain search in Domain Management in Calypso won't get bucketed into this abtest preemptively (this field is re-used in signup).

a8ci18n commented on client/components/domains/register-domain-step/index.jsx:
```diff
@@ -220,7 +221,10 @@ const RegisterDomainStep = React.createClass( {
-		const queryObject = getQueryObject( this.props );
+		const queryObject = getQueryObject( this.props ),
+			placeholder = ( this.props.isSignupStep && abtest( 'signupDomainsHeadline' ) === 'updated' )
+				? this.props.translate( 'Enter a name or keyword' )
```
Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Please enter a domain name or keyword.' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

yoavf replied:
@aidvu why thumbs down? - seems like a relevant suggestion to me.

aidvu replied:
Ah, I just re-read what it says. It is a good one. Thanks @yoavf.
